drivers/scheduler/dcos/marathon_ops.go

```diff
@@ -0,0 +1,110 @@
+package dcos
+
+import (
+	"regexp"
+	"sync"
+	"time"
+
+	marathon "github.com/gambol99/go-marathon"
+	"github.com/portworx/sched-ops/task"
+)
+
+// Ops is an interface to perform Marathon related operations
+type Ops interface {
+	ApplicationOps
+}
+
+// ApplicationOps is an interface to perform application operations
+type ApplicationOps interface {
+	// CreateApplication creates an application in Marathon
+	CreateApplication(*marathon.Application) (*marathon.Application, error)
+	// ValidateAppliation checks if the aplication is running and healthy
+	ValidateApplication(string) error
+	// DeleteApplication deletes the given application
+	DeleteApplication(string) error
+	// ValidateTerminatedApplication validates if the application is terminated
+	ValidateTerminatedApplication(string) error
+}
+
+var (
+	instance Ops
+	once     sync.Once
+)
+
+type marathonOps struct {
+	client marathon.Marathon
+}
+
+// MarathonClient returns a singleton instance of Marathon Ops type
+func MarathonClient() Ops {
+	once.Do(func() {
+		instance = &marathonOps{}
+	})
+	return instance
+}
+
+// Initialize Marathon client if not initialized
+func (m *marathonOps) initMarathonClient() error {
+	if m.client == nil {
+		marathonURL := "http://192.168.65.90:8080"
```

Comment:
How are we eventually going to find the marathon URL? (For e.g k8s gets it by the mounted px-account or the KUBE_CONFIG env variable).